src/devices/GrovePi/Sensors/LedBar.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Iot.Device.GrovePiDevice.Models;
+
+namespace Iot.Device.GrovePiDevice.Sensors
+{
+    /// <summary>
+    /// The led bar orientation
+    /// </summary>
+    public enum LedBarOrientation
+    {
+        RedToGreen = 0,
+        GreenToRed = 1
+    }
+
+    /// <summary>
+    /// LedBar class to support grove LedBar http://wiki.seeedstudio.com/Grove-LED_Bar/
+    /// </summary>
+    public class LedBar : ISensor<byte>
+    {
+        private GrovePi _grovePi;
+        private byte _level;
+        private LedBarOrientation _orientation;
+
+        /// <summary>
+        /// LedBar constructor
+        /// </summary>
+        /// <param name="grovePi">The GrovePi class</param>
+        /// <param name="port">The grove Port, need to be in the list of SupportedPorts</param>
+        public LedBar(GrovePi grovePi, GrovePort port) : this(grovePi, port, LedBarOrientation.GreenToRed)
```

Comment:
I feel like most of the sensors do not need to depend on the GrovePi and could just take I2c as an input - this way you could use them separately and they could live outside of GrovePi directory

Reply:
Unfortunately, none of those sensors are I2C. they all need GrovePi. Those sensors are purely Analogic or Digital sensors. For the Digital ones, the way GrovePi works won't make it easy to have it in a generic way. But for most of the digital sensors, there maybe a way to do it.
and in this case, based on your feedback on IGpioController, it would then be possible. 
I'll update this PR first like it is and will dig into implementing the IGpioController and see how I can make all the digital sensors (including the PWM ones) generic